comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -299,6 +330,70 @@ func (r *secretResolver) resolve(data []byte, origin string, notifyCb secrets.Re
 	return finalConfig, nil
 }
 
+// allowlistHandles restricts what config settings may be updated
+// tests can override this to exercise functionality: by setting this to nil, allow all handles
+// NOTE: Related feature to `authorizedConfigPathsCore` in `comp/api/api/apiimpl/internal/config/endpoint.go`
+var allowlistHandles = []string{"api_key"}
+
+func (r *secretResolver) processSecretResponse(secretResponse map[string]string, useAllowlist bool) {
+	// notify subscriptions about the changes to secrets
+	for handle, secretValue := range secretResponse {
+		// if allowlist is enabled and the handle is not contained in it, skip it
+		if useAllowlist && allowlistHandles != nil && !slices.Contains(allowlistHandles, handle) {
+			continue
+		}
+		oldValue := r.cache[handle]
+		// if value hasn't changed, don't send notifications
+		if oldValue == secretValue {
+			continue
+		}
```

Comment:
If a handle is used multiple times in the same configuration we would only notify on the first one.

Reply:
Fixed by moving the assignment to the cache to a second loop.